[EN-3815] Fix Not Applicable clearing in Foreign

christopherhuii opened the pull request:
## Description


## Checklist for Reviewer

- [ ] Review code changes
- [ ] Review changes for Database effects
- [ ] Verify change works in IE browser

More details about this can be found in [docs/review.md](docs/review.md)

suzubara commented on src/components/Section/Foreign/Activities/DirectActivity/DirectInterest.jsx:
```diff
@@ -49,107 +29,109 @@ export default class DirectInterest extends ValidationElement {
     this.update({
       RelinquishedNotApplicable: values,
-      Explanation: values.applicable ? this.props.Explanation : {}
+      Explanation: values.applicable ? this.props.Explanation : {},
```
It looks like this also needs to set `Relinquished` (the date itself isn't clearing out)

suzubara commented on src/components/Section/Foreign/Activities/IndirectActivity/IndirectInterest.jsx:
```diff
@@ -53,129 +32,131 @@ export default class IndirectInterest extends ValidationElement {
     this.update({
       SoldNotApplicable: values,
-      Explanation: values.applicable ? this.props.Explanation : {}
+      Explanation: values.applicable ? this.props.Explanation : {},
```
Same comment but for `Sold`

suzubara commented on src/components/Section/Foreign/Activities/RealEstateActivity/RealEstateInterest.jsx:
```diff
@@ -43,95 +27,98 @@ export default class RealEstateInterest extends ValidationElement {
+  updateSoldNotApplicable = (values) => {
     this.update({
-      SoldNotApplicable: values
+      Solid: values.applicable ? this.props.Solid : {},
```
typo! 😂 `Solid -> Sold`

christopherhuii replied:
...what on earth was going on with my brain when writing this 🙁

christopherhuii commented:
Real estate purchase: Fixed via #1740.
Foreign contacts: strange behavior with clearing: fixed in this PR
Foreign contacts: place of birth country field not clearing: fixed in this PR.

The fix for "place of birth country field not clearing" is temporary. The `<Dropdown />` component is persisting state and a thorough fix will probably need to touch multiple components outside of this scope. So instead, I moved around some code so that the values clear properly for the time being..

suzubara commented on src/components/Section/Foreign/Activities/DirectActivity/DirectInterest.jsx:
```diff
@@ -49,107 +29,109 @@ export default class DirectInterest extends ValidationElement {
     this.update({
       RelinquishedNotApplicable: values,
-      Explanation: values.applicable ? this.props.Explanation : {}
+      Relinquished: values.applicable ? this.props.Relinquished : {},
```
I'm not sure if this matters but the `Explanation` field is also hidden if relinquished is not applicable, so it might need to clear both of them?

suzubara commented on src/components/Section/Foreign/Activities/IndirectActivity/IndirectInterest.jsx:
```diff
@@ -53,129 +32,131 @@ export default class IndirectInterest extends ValidationElement {
     this.update({
       SoldNotApplicable: values,
-      Explanation: values.applicable ? this.props.Explanation : {}
+      Sold: values.applicable ? this.props.Sold : {},
```
same comment about `Explanation`
